app/controllers/restaurants_controller.rb

```diff
@@ -0,0 +1,45 @@
+class RestaurantsController < ApplicationController
+   def index
+     @restaurant = Restaurant.all
+   end
+ 
+   def show
+     @restaurant = Restaurant.find(params[:id])
+   end
+ 
+   def new
+     @restaurant = Restaurant.new
+   end
+ 
+   def edit
+     @restaurant = Restaurant.find(params[:id])
+   end
+ 
+   def create
+     @restaurant = Restaurant.create(restaurant_params)
+     if @restaurant.save 
+       redirect_to root_path
+       flash[:notice] = "Restaurant  was  saved  successfully"
+     else 

```

Comment:
I'm a little bit confused. Why is redirect_to characters_path? Shouldn't be restaurants_path because the resource is restaurant? I always confuse it!